client/state/imports/site-importer/actions.js

```diff
@@ -0,0 +1,265 @@
+	type: SITE_IMPORTER_VALIDATION_ERROR_SET,
+	message,
+} );
+
+export const startMappingSiteImporterAuthors = ( {
+	importerStatus,
+	site,
+	targetSiteUrl,
+} ) => dispatch => {
+	const singleAuthorSite = get( site, 'single_user_site', true );
+	const siteId = site.ID;
+	const { importerId } = importerStatus;
+
+	// WXR was uploaded, map the authors
+	if ( singleAuthorSite ) {
+		const currentUserData = user().get();
+		const currentUser = {
+			...currentUserData,
+			name: currentUserData.display_name,
+		};
+		const sourceAuthors = get( importerStatus, 'customData.sourceAuthors', [] );
+
+		// map all the authors to the current user
+		// TODO: when converting to redux, allow for multiple mappings in a single action
+		sourceAuthors.forEach( author => mapAuthor( importerId, author, currentUser ) );
+
+		// Check if all authors are mapped before starting the import.
+		defer( () => {
+			const newState = get( getImporterState(), [ 'importers', importerId ] );
+			const areAllAuthorsMapped = every(
+				get( newState, 'customData.sourceAuthors', [] ),
+				'mappedTo'
+			);
+
+			if ( areAllAuthorsMapped ) {
+				startImporting( newState );
+
+				dispatch(
+					recordTracksEvent( 'calypso_site_importer_map_authors_single', {
+						blog_id: siteId,
+						site_url: targetSiteUrl,
+					} )
+				);
+			}
+		} );
+	} else {
+		startMappingAuthors( importerId );
+
+		recordTracksEvent( 'calypso_site_importer_map_authors_multi', {
+			blog_id: siteId,
+			site_url: targetSiteUrl,
+		} );
+	}
+};
+
+export const importSite = ( {
+	engine,
+	importerStatus,
+	params,
+	site,
+	supportedContent,
+	targetSiteUrl,
+	unsupportedContent,
+} ) => dispatch => {
+	const siteId = site.ID;
+	const trackingParams = {
+		blog_id: siteId,
+		site_url: targetSiteUrl,
+		supported_content: supportedContent
+			.slice( 0 )
+			.sort()
+			.join( ', ' ),
+		unsupported_content: unsupportedContent
+			.slice( 0 )
+			.sort()
+			.join( ', ' ),
+		site_engine: engine,
+	};
+
+	dispatch(
+		withAnalytics(
+			recordTracksEvent( 'calypso_site_importer_start_import_request', trackingParams ),
+			startSiteImporterImport()
+		)
+	);
+
+	wpcom.wpcom.req
+		.post( {
+			path: `/sites/${ siteId }/site-importer/import-site?${ stringify( params ) }`,
+			apiNamespace: 'wpcom/v2',
+			formData: [
+				[ 'import_status', JSON.stringify( toApi( importerStatus ) ) ],
+				[ 'site_url', targetSiteUrl ],
+			],
+		} )
+		.then( response => {
+			// At this point we're assuming that an import is going to happen
+			// so we set the user's editor to Gutenberg in order to make sure
+			// that the posts aren't mangled by the classic editor.
+			if ( 'engine6' === engine ) {
+				dispatch( setSelectedEditor( siteId, 'gutenberg' ) );
+			}
+
+			dispatch(
+				withAnalytics(
+					recordTracksEvent( 'calypso_site_importer_start_import_success', trackingParams ),
+					siteImporterImportSuccessful( response )
+				)
+			);
+
+			defer( () => {
```

Comment:
The `defer` call here is not needed. It's useful to prevent dispatching a Flux action while another Flux dispatch callback is still running. But that's not happening here. The code is executed within a promise's `then` callback, so it's guaranteed there is no Flux dispatch happening up the call stack.